tests/python/direct/test_narrow_precision.py

```diff
@@ -67,6 +67,47 @@ def nvfp4_quantize_with_te(input_tensor):
         return None
 
 
+# https://github.com/NVIDIA/TransformerEngine/blob/d126cdd6c0a8d6ce0419dcf1ffe027ef7aadf7e8/tests/cpp/operator/test_cast_nvfp4_transpose.cu#L56-L68
+def compute_nvfp4_global_scale(tensor):
+    """Compute global scale factor for NVFP4 quantization.
+
+    Args:
+        tensor: Input tensor to compute scale for
+        device: Device to create constant tensors on
+
+    Returns:
+        Global scale as (FLOAT8_E4M3_MAX * FLOAT4_E2M1_MAX) / max(abs(tensor))
+    """
+    amax = torch.max(torch.abs(tensor)).to(torch.float32)
+    x_global_scale = (FLOAT8_E4M3_MAX * FLOAT4_E2M1_MAX) / amax
+    x_global_scale = torch.clamp(x_global_scale, max=torch.finfo(torch.float32).max)
+
+    # Handle edge cases: zero input or invalid scale
+    invalid_mask = (amax == 0.0) | (x_global_scale == 0.0)
+    x_global_scale = torch.where(
+        invalid_mask, torch.tensor(1.0, device="cuda"), x_global_scale
```

Comment:
super nitpick: we can just use `tensor.device` to avoid device mismatch, which isn't a problem for scalar tensor... so no big deal.

Also, `torch.where(invalid_mask, 1.0, x_global_scale)` should work?!